docs/architecture/adr-042-state-sync.md

```diff
@@ -0,0 +1,225 @@
+with a verified block header.
+
+A similar solution was implemented by Binance in
+[#3594](https://github.com/tendermint/tendermint/pull/3594)
+based on their initial implementation in
+[PR #3243](https://github.com/tendermint/tendermint/pull/3243)
+and [some learnings](https://docs.google.com/document/d/1npGTAa1qxe8EQZ1wG0a0Sip9t5oX2vYZNUDwr_LVRR4/edit).
+Note this still requires the honest majority peer assumption.
+One major advantage of the warp-sync approach is that chunks can be compressed before sending.
+
+Ideally State Sync in Tendermint would not require an honest majority of
+peers, and instead would rely on the standard light client security,
+meaning state chunks would be verifiable against the merkle root of the
+state. This property tends to come more naturally to the Lazy approach
+since it tends to use the existing structure of the tree (ie. keys or
+nodes) rather than state-sync specific chunks. Such a design for
+tendermint was originally tracked in
+[#828](https://github.com/tendermint/tendermint/issues/828).
+
+### Analysis of Lazy vs Eager
+
+Lazy vs Eager have more in common than they differ. They all require
+reactors on the tendermint side, a set of ABCI messages and a method for
+serializing/deserializing snapshots facilitated by a SnapshotFormat.
+
+The biggest difference between Lazy and Eager proposals is in the
+read/write patterns necessitated by when serving a snapshot chunk.
+Specifically, Lazy State Sync random reads to the underlying data
+structure while Eager can optimize for sequential reads.
+
+This distinctin between approaches was demonstrated by Binance's [ackratos](https://github.com/ackratos) in their
+implementation of [Lazy State sync](https://github.com/tendermint/tendermint/pull/3243), The [analysis](https://docs.google.com/document/d/1npGTAa1qxe8EQZ1wG0a0Sip9t5oX2vYZNUDwr_LVRR4/) of the performance, and
+follow up implementation of [Warp Sync](http://github.com/tendermint/tendermint/pull/3594).
+
+#### Compairing Security Models
+One proposed difference between Lazy and Eager state sync is the
+security model. It had been suggested that Lazy State Sync could use
+light client validation while Warp Sync was unsuitable for public
+networks as it relies on the majority of peers to produce a manifest.
+This manifest would require complete syncing before validation of the
+data against the merkle root were possible. This could create DDOS
+attack vector in which a peer publishes an invalid manifest and making
+peers download invalid data. For this reason, WarpSync is faster but
+less safety-efficient; It as it takes longer to realize that data is
+invalid.
+
+However, upon analysis of the approaches with the orthogonal axis of
+eager/lazy and chunk/snapshot validation suggest that a version of
+WarpSync with per chunk light client validation might get the optimal
+performance and safety.
+
+## Decision: Eager StateSync With Per Chunk Light Client Validation
```

Comment:
Would this require bringing manifest hash into consensus?

Reply:
Will we only require validator nodes to compute and validate this hash?

Reply:
No it would not, the application state would be validated against the merkle root produced from light client sync.